drivers/lvutil.py

```diff
@@ -138,6 +138,34 @@ def __exit__(self, exc_type, value, traceback):
             self.lock.release()
 
 
+LVM_RETRY_ERRORS = [
+    "Incorrect checksum in metadata area header"
+]
+
+
+def lvmretry(func):
+    def check_exception(exception):
+        retry = False
+        for error in LVM_RETRY_ERRORS:
+            if error in exception.reason:
+                retry = True
+        return retry
+
+    def decorated(*args, **kwargs):
+        retry = True
+        while retry:
+            retry = False
+            try:
+                func(*args, **kwargs)
```

Comment:
Once a counter is added I think we should log a message here if we succeed with counter > 1. Then we can have a healthcheck in XenRT to detect this condition and track how likely it is.